File: bidding-engine/src/main/java/com/omnibus/bidding/events/NoopDomainEventPublisher.java
package com.omnibus.bidding.events;

import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
import org.springframework.stereotype.Component;

@Component
@ConditionalOnProperty(name = "omnibus.kafka.enabled", havingValue = "false")
public class NoopDomainEventPublisher implements DomainEventPublisher {

    @Override
    public void publishRecommendationProduced(RecommendationProducedEvent event) {
    }

    @Override
    public void publishRuleUpdated(RuleUpdatedEvent event) {
    }
}

File: services/bidding-engine/src/main/kotlin/pl/writeonly/omnibus/bidding/events/NoopDomainEventPublisher.kt
package pl.writeonly.omnibus.bidding.events

import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
import org.springframework.stereotype.Component

@Component
@ConditionalOnProperty(name = ["omnibus.kafka.enabled"], havingValue = "false")
class NoopDomainEventPublisher : DomainEventPublisher {
    override fun publishRecommendationProduced(event: RecommendationProducedEvent) {
        // intentionally no-op
    }

    override fun publishRuleUpdated(event: RuleUpdatedEvent) {
        // intentionally no-op
    }
}

